File: Competitive Programming 3/Section 1.4/Palidromes/353 - Pesky Palindromes/C++/main.cpp
#include<bits/stdc++.h>

using namespace std;

bool isPalindrome(const string &s){
    int lim = s.size()>>1;
    for(int i=0;i<lim;i++)
        if(s[i]!=s[s.size()-i-1])
            return false;
    return true;
}

int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);
    string s;
    while(cin>>s){
        set<string> differentPs;
        for(int i=0;i<s.size();i++){
            for(int j=i;j<s.size();j++){
                auto substring = s.substr(i,j-i+1);
                if(isPalindrome(substring))
                    differentPs.insert(substring);
            }
        }
        cout << "The string '" << s << "' contains " << differentPs.size() << " palindromes.\n"; 
    }
    return 0;
}

File: Competitive Programming 3/Section 1.4/Palidromes/353 - Pesky Palindromes/Java/Main.java
import java.util.Scanner;
import java.util.HashSet;

public class Main{
    static Scanner s = new Scanner(System.in);
    static boolean isPalindrome(String s){
        int lim = s.length()>>1;
        for(int i=0;i<lim;i++)
            if(s.charAt(i)!=s.charAt(s.length()-i-1))
                return false;
        return true;
    }
    public static void main(String[]args){
        while(s.hasNextLine()){
            String line = s.nextLine();
            HashSet<String> differentPs = new HashSet<String>();
            for(int i=0;i<line.length();i++){
                for(int j=i;j<line.length();j++){
                    String substring = line.substring(i,j+1);
                    if(isPalindrome(substring))
                        differentPs.add(substring);
                }
            }
            System.out.printf("The string '%s' contains %d palindromes.\n",line,differentPs.size());
        }
    }
}